### PR TITLE
add support for changing the weave peer connection limit

### PR DESCRIPTION
**Pull request by polarbizzle**
Weave limits the number of connections between peers and the default is 30.
As described here: https://github.com/weaveworks/weave/blob/ce9c5162bd94018ad5caca2c914cb5780727147c/site/kubernetes/kube-addon.md
Setting the CONN_LIMIT environment variable controls this setting. Adding a field in the
WeaveNetworkingSpec and adding the env var to the associated templates lets users
have clusters larger than 30 nodes and still use weave's peer discovery

The weave authors have indicated that 100 is not an unreasonable setting.
weaveworks/weave#1621 (comment)

### CODE REVIEW

File: upup/pkg/fi/cloudup/bootstrapchannelbuilder.go [hunk 1]
```diff
@@ -388,8 +388,8 @@ func (b *BootstrapChannelBuilder) buildManifest() (*channelsapi.Addons, map[stri
 
 	if b.cluster.Spec.Networking.Weave != nil {
 		key := "networking.weave"
-		// 2.1.3-kops.1 = 2.1.3, kops packaging version 1.
-		version := "2.1.3-kops.1"
+		// 2.1.3-kops.2 = 2.1.3, kops packaging version 2.
+		version := "2.1.3-kops.2"
```
Comment (polarbizzle) [inline upup/pkg/fi/cloudup/bootstrapchannelbuilder.go]: @chrislovecnm I believe this is the version you mentioned?

Comment (chrislovecnm) [reply upup/pkg/fi/cloudup/bootstrapchannelbuilder.go]: Sorry if I missed the update ...

**Comment by rhefner1** [comment]
Wow, we just hit this issue this afternoon trying to scale up nodes. Haven't tried it yet, but this seems to explain the exact problem we had. I wish this were far more prominent in the weave docs, so thanks @polarbizzle for the PR.

**Comment by zacblazic** [comment]
Ah @polarbizzle you beat me to this!

We had a bit of a scare when our 31st and 32nd nodes became peers of each other instead of the rest of the cluster.

Also note that weave [have bumped their default connection limit to 100](https://github.com/weaveworks/weave/pull/3234).

**Pull request merged**